gcc/rust/lint/rust-lint-scan-deadcode.h

```diff
@@ -46,20 +46,34 @@ class ScanDeadcode : public MarkLiveBase
   void visit (HIR::Function &function) override
   {
     HirId hirId = function.get_mappings ().get_hirid ();
-    if (live_symbols.find (hirId) == live_symbols.end ())
-      {
-	rust_warning_at (function.get_locus (), 0,
-			 "function is never used: %<%s%>",
-			 function.get_function_name ().c_str ());
-	return;
-      }
+    warning (hirId, function.get_locus (), "function",
```

Comment:
I think i would prefer a function

warning_unused (...) so you don't need to pass the argument "used"

Reply:
actually ignore this comment for now let me think about it and we can change in a future PR